### PR TITLE
feature(api): add service for jobs status

### PR DESCRIPTION
**Pull request by Dyostiq**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  [API_EVENT_KINDS.project__protectedAreas__failed__v1__alpha]: undefined,
+  [API_EVENT_KINDS.project__protectedAreas__finished__v1__alpha]: undefined,
+  [API_EVENT_KINDS.project__protectedAreas__submitted__v1__alpha]: undefined,
+  [API_EVENT_KINDS.scenario__costSurface__costUpdateFailed__v1_alpha1]:
+    JobStatus.failure,
+  [API_EVENT_KINDS.scenario__costSurface__finished__v1_alpha1]: JobStatus.done,
+  [API_EVENT_KINDS.scenario__costSurface__shapeConversionFailed__v1_alpha1]:
+    JobStatus.failure,
+  [API_EVENT_KINDS.scenario__costSurface__shapeConverted__v1_alpha1]:
+    JobStatus.running,
```
Comment (kgajowy) [inline api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts:60]: Seems to cover existing https://github.com/Vizzuality/marxan-cloud/pull/253/files#diff-d0dcb799abb6600133ff14d39c366ce3ccad3324da0c01e42732a9745f4f1682

One of those should be enough (the linked one tho seems to be more readable ;) )

File: api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      projects.id AS "projectId",
+      api_events.kind
+    FROM
+      api_events
+      INNER JOIN scenarios ON api_events.topic = scenarios.id
+      INNER JOIN projects ON projects.id = scenarios.project_id
+      CROSS JOIN LATERAL SUBSTRING(
+        api_events.kind
+        FROM
+          'scenario.#"[^.]*#"%' FOR '#'
```
Comment (kgajowy) [inline api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts:20]: That looks scary, what it exactly matches?

**Review by kgajowy — APPROVED** [review]

File: api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts (hunk 1 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts:60]: It contains much less events (e.g. missing `shapeConversionFailed`) and does not have the compiler enforcement that I wanted.

File: api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts (hunk 2 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts:20]: it's just a regular expression
https://www.postgresql.org/docs/9.3/functions-matching.html#FUNCTIONS-SIMILARTO-REGEXP
matches everything what is not a dot after `scenario.` and substrings it

**Pull request merged**